File: devops/Gestion_Reservation/reservation-service/src/main/java/cheikh/beidja/reservationservice/entities/Reservation.java
package cheikh.beidja.reservationservice.entities;

import cheikh.beidja.reservationservice.model.Resource;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Reservation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nom;
    private String context;
    private Date date;
    private int duration;
    private Long resourceId;
    @ManyToOne
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    private Person person;
    @Transient
    private Resource resource;
}

File: angular-front-app/src/Reservation.ts
import {Resource} from "./Resource";
import {Person} from "./Person";

export interface Reservation{
  id:number;
  nom:string;
  context:string;
  date:Date;
  duration:number;
  resource:Resource;
  person:Person
  personId:number;
  resourceId:number;



}
